datalad_next/constraints/tests/test_special_purpose.py

```diff
@@ -356,3 +356,14 @@ def test_EnsureDataset(tmp_path):
     assert EnsureDataset(
         installed=False).short_description() == \
         '(path to) a non-existing Dataset'
+
+    # smoke test for idcheck:
+    assert EnsureDataset(idcheck=True)(ds).ds == ds
+    assert EnsureDataset(idcheck=False)(ds).ds == ds
+    # unset the dataset ID to test whether an ID check would raise, but
+    # bring it back later in case future tests need it
+    id = ds.config.get('datalad.dataset.id')
+    ds.config.unset('datalad.dataset.id')
+    with pytest.raises(NoDatasetFound):
+        EnsureDataset(idcheck=True)(tmp_path)
+    ds.config.set('datalad.dataset.id', id)
```

Comment:
The dataset is temporary, there is no need to undo the changes. If you want to be more specific, you could use `scope='branch'` for `ds.config.unset()`.

If you want to be extra-safe re config leakage, use the `datalad_cfg` fixture. But all tests in `-next` are automatically verified to not change the persistent configuration. So if this one passes, we are safe.

Reply:
thx for the tip!

Reply:
I've changed the scope to branch.